py/loss_functions.py

```diff
@@ -29,8 +29,9 @@ def loss_metrics(x: list) -> pd.Series:
     :rtype: pd.Series
     """
     # Extract parameters and generate reform DataFrame.
-    senior, child, dis_1, dis_2, dis_3 = x[:5]
-    regions = np.array(x[5:])
+    senior = x[0]
+    child, dis_1, dis_2, dis_3 = x[2:6]
+    regions = np.array(x[6:])
     reform_df = set_ubi(
         reform_base_df, budget, senior, child, dis_1, dis_2, dis_3, regions
```

Comment:
```suggestion
        reform_base_df, budget, senior, child, dis_1, dis_2, dis_3, regions, verbose=True
```
(if you want to print out the adult amount)

Reply:
> Or are you looking for a function to extract the adult UBI, not just print it? @nikhilwoodruff is there a clean way to pull out the adult amount calculation from `set_ubi` in a way that can be used here?

Ideally I'd like to extract the adult amount from set_ubi so that I can set x[1] = adult_amount before creating and printing the output_dict.

Reply:
> Uploaded the notebook now - optimization runs (very fast), but the amount for seniors isn't obeying the bounds that were passed.

Weird. I can't find anything about this issue. If we can reproduce could be worth filing an issue in the scipy repo. In the meantime, could you make the function value `np.inf` (or some big constant) for scenarios that violate the constraints? Seems like this would be necessary to respect the adult amount bounds anyway (though that doesn't seem to be an issue).

Could you prioritize this over getting the adult amount in the print message?

Reply:
I've managed to get the optimization working and I've uploaded a notebook showing two different optimizations

The adult amount was being calculated but then not being used in the actual optimization, which was causing problems (in addition to an error with ordering the output_dict), so I created a function in calc_ubi.py to extract the adult amount directly and then feed it into the optimization. Currently the optimization works so long as you don't fix any variables (see the third example in the notebook where I try to fix the regional supplements). You can see from the output that it progressed somewhat (with the amounts obeying the bounds) before exiting with the error.

Calling get_data() in optimize.py and the extract() function are both areas which could probably be simplified by moving them to just optimize.py instead of having them in loss_functions.py too, but the optimization still runs fast.